spackbot/pr_run_pipeline.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import logging
+import random
+import requests
+import re
+import os
+
+from .helpers import found, alias_regex
+from gidgethub import routing
+
+
+logger = logging.getLogger(__name__)
+router = routing.Router()
+
+# We can only make the request with a GITLAB TOKEN
+GITLAB_TOKEN = os.environ.get("GITLAB_TOKEN")
+
+
+async def run_pipeline(event, gh):
+    """
+    Make a request to re-run a pipeline.
+    """
+    # Get the pull request number
+    number = event.data["issue"]["pull_request"]["url"].split("/")[-1]
```

Comment:
isn't the PR number already an int in `event.data["issue"]["pull_request"]["number"]`?

Reply:
Nope. But you can get it from the end of the url there.

```
    "pull_request": {
      "url": "https://api.github.com/repos/vsoch/spack/pulls/15",
      "html_url": "https://github.com/vsoch/spack/pull/15",
      "diff_url": "https://github.com/vsoch/spack/pull/15.diff",
      "patch_url": "https://github.com/vsoch/spack/pull/15.patch"
    },
```

Reply:
I think the idea is that the developer is expected to use the PR url to get metadata about the PR. But we just need the number so it's easier to just grab it from the url.

Reply:
Yeah I agree this is better.